python/lsst/qa/explorer/classifyObjects.py

```diff
@@ -0,0 +1,371 @@
+        camera_name = dataRef.getButler()._getDefaultMapper().getCameraName()
+        if camera_name != "hsc":
+            raise NotImplementedError
+
+        # To Do: DM-14539 - Move this to somewhere else
+        # put this into /datasets/hsc then the butler will get it
+        # Add classifier_pickle to obs_base datasets, copy deepCoadd_skyMap.
+        clfFilenameMorph = "/home/sr525/starGalaxy/clfIsotonicMorph.pickle"
+        with open(clfFilenameMorph, "rb") as f:
+            filters = pickle.load(f)
+
+        filtersInCat = set(cat.columns.get_level_values(0))
+        if filtersInCat != set(filters):
+            raise RuntimeError("Not all filters required in file")
+
+        # addMagnitdues will probably be removed later as it will hopefully be superseded by Tim
+        cat = addMagnitudes(cat, dataRef, filters)
+        cat = self.run(cat, filters)
+
+        cat = ParquetTable(dataFrame=cat)
+        dataRef.put(cat, "deepCoadd_sg")
+
+    def run(self, cat, filters):
+        """Add colour based classification, morphology only based classification and their associated flags.
+
+        Parameters
+        ----------
+        cat : 'Pandas data frame'
+        filters : 'list'
+            List of the filters for the camera used
+
+        Returns
+        -------
+        Data frame with the two classification and two flag columns added and a new index containing the
+        columns required by the classifier
+
+        Notes
+        -----
+        For the morphology only classifier the features used are psf - cmodel in r, i and z, m, b, seeing in
+        g, r, i, z and Y, flux/flux error in g, r, i, z and Y, psf - cmodel error in r, i and z
+        For the colour and morphology classifier the features used are g - r, r - i, i - z, z - Y (all using
+        cmodel), psf - cmodel in g, r, i, z and Y, m, b, flux / flux error in g, r, i, z and Y, seeing in g,
+        r, i, z and Y, psf - cmodel error in g, r, i, z and Y
+        The columns must be used in this order
+
+        Adds columns to the input table for morphology only classification, morph + color classification and
+        their associated flags, adds a new index to the file that contains the columns used by the
+        classifier.
+        """
+        # Add required columns
+        cat = addSeeing(cat, filters)
+        cat = addColors(cat, filters)
+        cat = addMB(cat, filters)
+        cat = addSNFlux(cat, filters)
+
+        # To Do: DM-14539 - Move this to somewhere else
+        # put this into /datasets/hsc then the butler will get it
+        # Add classifier_pickle to obs_base datasets, copy deepCoadd_skyMap.
+        clfFilenameMorph = "/home/sr525/starGalaxy/clfIsotonicMorph.pickle"
+        with open(clfFilenameMorph, "rb") as f:
+            filters = pickle.load(f)
+            clfMorph = pickle.load(f)
+            colsToUseMorph = pickle.load(f)
```

Comment:
I strongly suggest you read this object and the other pickle (clfIsotonicColour.pickle) in `runDataRef` and pass both objects as arguments to `run`. That is likely how things will be after DM-14539: these products will be read using the butler, and thus will be read in `runDataRef` and passed to `run`.

It also gives you a natural place to document what these objects are (as `run` parameters), though that is not the primary motivation.